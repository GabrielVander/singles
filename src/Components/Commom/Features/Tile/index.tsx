import React, {ReactElement} from 'react';
import {Anchor, Box, Paragraph} from 'grommet';

interface TileProps {
    children?: React.ReactNode;
    action?: string;
    summary?: string;
    direction?: 'row' | 'column' | 'row-responsive' | 'row-reverse' | 'column-reverse' | undefined;
}

const Tile = ({ children, action, summary, direction }: TileProps): ReactElement => (
    <Box basis="medium" align="center" margin="medium">
        <Box height="small" align="center" justify="center" direction={direction}>
            {children}
        </Box>
        <Paragraph size="large" textAlign="center">
            {summary}
        </Paragraph>
        <Box align="center">
            <Anchor href="/" alignSelf="center">
                {action}
            </Anchor>
        </Box>
    </Box>
);

export default Tile;
