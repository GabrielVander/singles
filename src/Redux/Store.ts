import {createStore} from "redux";
import rootReducer from "./Reducers";

class Store {
    static INITIAL_GLOBAL_STATE = {};
    // @ts-ignore
    static store = createStore(rootReducer, Store.INITIAL_GLOBAL_STATE, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());
}

export default Store;
