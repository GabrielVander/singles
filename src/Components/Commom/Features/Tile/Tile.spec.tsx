import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import Store from "../../../../Redux/Store";
import {BrowserRouter} from "react-router-dom";
import React from "react";
import Tile from "./index";

test('matches snapshot', () => {
    const component = render(
        <Provider store={Store.store}>
            <BrowserRouter>
                <Tile/>
            </BrowserRouter>
        </Provider>
    );
    expect(component).toMatchSnapshot();
});
