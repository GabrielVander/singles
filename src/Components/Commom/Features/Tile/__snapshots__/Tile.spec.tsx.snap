// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="StyledBox-sc-13pk1d4-0 bXRCHS"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 cHQFNL"
        />
        <p
          class="StyledParagraph-tbetod-0 fNpWwn"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 dvIeIL"
        >
          <a
            class="StyledAnchor-sc-1rp7lwl-0 lcrrzR"
            href="/"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="StyledBox-sc-13pk1d4-0 bXRCHS"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 cHQFNL"
      />
      <p
        class="StyledParagraph-tbetod-0 fNpWwn"
      />
      <div
        class="StyledBox-sc-13pk1d4-0 dvIeIL"
      >
        <a
          class="StyledAnchor-sc-1rp7lwl-0 lcrrzR"
          href="/"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
